tests/NuGet.Services.AzureSearch.Tests/SearchService/SearchResponseBuilderFacts.cs

```diff
@@ -357,6 +360,88 @@ public void UsesVersionSpecificDownloadCount()
                 Assert.Equal(4, response.Data[0].DownloadCount);
             }
 
+            [Fact]
+            public void SortDownloadsAscUsingAuxilaryFile()
+            {
+                var mockAuxilaryDataDownloads = new Dictionary<string, long>();
+                var rnd = new Random();
+                for (int i = 0; i < _fiveSearchResults.Count; ++i)
+                {
+                    var newDoc = _fiveSearchResults.Results[i];
+                    newDoc.Document.PackageId += i;
+                    newDoc.Document.TotalDownloadCount += i;
+                    newDoc.Document.Title += i;
+
+                    // Set random download count in the Auxilary file for this package
+                    mockAuxilaryDataDownloads.Add(newDoc.Document.PackageId, rnd.Next(0, int.MaxValue));
```

Comment:
I think using random here will work as expected, but it scares me a little. Using random values means that these tests are not deterministic and may be flaky. Do you think that the randomness here covers more cases than having a deterministic test? If so, could we instead add these cases manually such that they are deterministic?